main.tf

```diff
@@ -1,13 +1,204 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled                 = module.this.enabled
+  create_log_bucket       = var.bucket_id == null ? true : false
+  bucket_id               = var.bucket_id != "" ? var.bucket_id : module.ssm_patch_log_s3_bucket.bucket_id
+  default_allowed_actions = ["s3:GetObject", "s3:PutObject", "s3:PutObjectAcl", "s3:GetEncryptionConfiguration"]
+}
+
+module "ssm_patch_log_s3_bucket" {
+  count                  = local.create_log_bucket ? 1 : 0
+  source                 = "cloudposse/s3-bucket/aws"
+  version                = "0.38.0"
+  acl                    = "private"
+  versioning_enabled     = false
+  allowed_bucket_actions = var.ssm_bucket_policy != null ? [] : local.default_allowed_actions
+  policy                 = var.ssm_bucket_policy
+  context                = module.this.context
+}
+
+resource "aws_ssm_maintenance_window" "scan_window" {
+  count    = module.this.enabled ? 1 : 0
+  name     = "${module.this.id}-scan-window"
+  schedule = var.scan_maintenance_window_schedule
+  duration = var.maintenance_window_duration
+  cutoff   = var.maintenance_window_cutoff
+}
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+resource "aws_ssm_maintenance_window_task" "task_scan_patches" {
+  count            = module.this.enabled ? 1 : 0
+  window_id        = aws_ssm_maintenance_window.scan_window[0].id
+  task_type        = "RUN_COMMAND"
+  task_arn         = "AWS-RunPatchBaseline"
+  priority         = var.task_scan_priority
+  service_role_arn = var.service_role_arn
+  max_concurrency  = var.max_concurrency
+  max_errors       = var.max_errors
+
+  targets {
+    key    = "WindowTargetIds"
+    values = aws_ssm_maintenance_window_target.target_scan.*.id
+  }
+
+  task_invocation_parameters {
+    run_command_parameters {
+      parameter {
+        name   = "Operation"
+        values = ["Scan"]
+      }
+      parameter {
+        name   = "RebootOption"
+        values = ["NoReboot"]
+      }
+      output_s3_bucket     = local.bucket_id
+      output_s3_key_prefix = "scaning"
+
+      service_role_arn = var.role_arn_for_notification
+      dynamic "notification_config" {
+        for_each = var.enable_notification_scan ? [1] : []
+        content {
+          notification_arn    = var.notification_arn
+          notification_events = var.notification_events
+          notification_type   = var.notification_type
+        }
+      }
+    }
   }
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+resource "aws_ssm_maintenance_window_target" "target_scan" {
+  count         = module.this.enabled ? 1 : 0
+  window_id     = aws_ssm_maintenance_window.scan_window[0].id
+  resource_type = "INSTANCE"
+
+  dynamic "targets" {
+    for_each = length(var.scan_maintenance_windows_targets) > 0 ? var.scan_maintenance_windows_targets : []
```

Comment:
why not just

```
for_each = var.scan_maintenance_windows_targets
```

If it's an empty array, it will do the same as your code.
If it's any other type (including `null`), it will trows errors (which is what we want)